Alerting Docs: Update Alert rule and Notification Templating (#94069)
grafana/grafana: docs/sources/alerting/configure-notifications/template-notifications/manage-notification-templates.md

description: Create notification templates to sent to your contact points
keywords:
  - grafana
  - alerting
  - notifications
  - templates
  - create templates
  - edit templates
  - delete templates
labels:
  products:
    - cloud
    - enterprise
    - oss
title: Manage notification templates
menuTitle: Manage templates
weight: 101
refs:
  notification-template-reference:
    - pattern: /docs/grafana/
      destination: /docs/grafana/<GRAFANA_VERSION>/alerting/configure-notifications/template-notifications/reference/
    - pattern: /docs/grafana-cloud/
      destination: /docs/grafana-cloud/alerting-and-irm/alerting/configure-notifications/template-notifications/reference/
  notification-template-examples:
    - pattern: /docs/grafana/
      destination: /docs/grafana/<GRAFANA_VERSION>/alerting/configure-notifications/template-notifications/language/
    - pattern: /docs/grafana-cloud/
      destination: /docs/grafana-cloud/alerting-and-irm/alerting/configure-notifications/template-notifications/language/
---

# Manage notification templates

> In contact points, you can select notification templates to decide the notification messages sent to their receivers.

Reviewer: ```suggestion
In contact points, you can select notification templates to customize the notification messages sent.
```

By default, Grafana provides a template for the notification title (`default.title`) and a template for the notification message (`default.message`). Both default templates display common alert details.

You can also create custom templates to customize the content and format of notification messages, which can then be applied to one or more contact points.

This documentation provides step-by-step instructions for selecting templates in contact points, previewing templates, and creating custom templates using the Grafana UI.

## Select a notification template for a contact point

> To add an existing notification template to your contact point, follow these steps:

Reviewer: ```suggestion
To add an existing notification template to your contact point, complete the following steps.
```

> 1. Click an existing contact point or create a new one

Reviewer: ```suggestion
1. Click an existing contact point or create a new one.
```

   For example, if you are creating an email contact point integration, click **Message** or **Subject**.

1. Click **Edit**.
   A dialog box opens where you can select templates.
1. Click **Select existing template** to select a template and [preview](#preview-a-notification-template) it using the default payload.

   You can also copy the selected template and use it in the custom tab.

1. Click **Enter custom message** to customize and edit the field directly. Note that the title changes depending on the field you are editing.

1. You can switch between the two tabs to access the list of available templates and copy them across to the customized version.

1. Click **Save contact point**.

## Create a notification template

Create notification templates to customize notification messages and reuse them in contact points.

Your notification template name must be unique. You cannot have two templates with the same name in the same notification template or in different notification templates. Avoid defining templates with the same name as default templates, such as: `__subject`, `__text_values_list`, `__text_alert_list`, `default.title` and `default.message`.

> To create a notification template in Grafana, complete the following steps:

Reviewer: ```suggestion
To create a notification template in Grafana, complete the following steps.
```

1. Click **Alerts & IRM** -> **Contact points**.
1. Click the **Notification Templates** tab and then **+ Add notification template**.

1. Enter a name for the notification template.

1. Write the content of the template in the content field.

1. Save your changes.

   If `{{ define }}` is not included in the content, `{{ define "<NOTIFICATION_TEMPLATE_NAME>" }}` and `{{ end }}` is automatically added to the start and end.

> To create a notification template that contains more than one template:

Reviewer: ```suggestion
To create a notification template that contains more than one template, complete the following steps.
```

1. Click **+ Add notification template**.

> 2. Enter a name for the notification template.

Reviewer: ```suggestion
1. Enter a name for the notification template.
```

> 3. Write each template in the Content field, including `{{ define "name-of-template" }}` and `{{ end }}` at the start and end of each template.

Reviewer: ```suggestion
1. Write each template in the Content field, including `{{ define "name-of-template" }}` and `{{ end }}` at the start and end of each template.
```

> 4. Save your changes.

Reviewer: ```suggestion
1. Save your changes.
```
